lib/buffer.js

```diff
@@ -64,7 +64,17 @@ function alignPool() {
  * much breakage at this time. It's not likely that the Buffer constructors
  * would ever actually be removed.
  **/
+var newBufferWarned = false;
 function Buffer(arg, encodingOrOffset, length) {
+  if (!new.target && !newBufferWarned) {
+    newBufferWarned = true;
+    process.emitWarning(
+      'Using Buffer without `new` will soon stop working. ' +
+      'Use `new Buffer()`, or preferably ' +
+      '`Buffer.from()`, `Buffer.allocUnsafe()` or `Buffer.alloc()` instead.',
```

Comment:
It's ok to keep allocUnsafe here, but the order should be different, I think: from, alloc, allocUnsafe.